.github/workflows/ci-examples.yml

```diff
@@ -52,14 +48,12 @@ jobs:
         with:
           java-version: '8.0.302'
           distribution: temurin
-      - name: Cache Gradle Home files
-        uses: actions/cache@v3.0.11
-        continue-on-error: true
-        with:
-          path: ~/.gradle/caches
-          key: ${{ runner.os }}-gradle-home-examples-${{matrix.gradle_args}}_check-${{ hashFiles('**/*.gradle') }}
       - name: Clear existing docker image cache
         run: docker image prune -af
+      - name: Setup Gradle Build Action
+        uses: gradle/gradle-build-action@v2
+        with:
+          gradle-home-cache-cleanup: true
```

Comment:
I had to look up the effect of it because I think the name is confusing but it looks like a good idea. I wonder why it is not a default?

https://github.com/gradle/gradle-build-action#removing-unused-files-from-gradle-user-home-before-saving-to-cache